scss/_navbar.scss

```diff
@@ -77,9 +77,8 @@
   }
 
   .dropdown-menu {
-    position: static !important;
+    position: static;
```

Comment:
I'm on your branch `.v4-dropdown-navbar` and made two pens:
- one with the last commit: [e836f5](https://codepen.io/zalog/pen/OgbVzE)
- the other with one commit before the last commit: [7d3368](https://codepen.io/zalog/pen/bRBdLm)

The problem is that with our last commit:
```
.navbar-expand {
  @each $breakpoint in map-keys($grid-breakpoints) {
    $next: breakpoint-next($breakpoint, $grid-breakpoints);
    $infix: breakpoint-infix($next, $grid-breakpoints);

    &#{$infix} {
      @include media-breakpoint-down($breakpoint) {
        .dropdown-menu {
          position: static;
          float: none;
        }
...
```
the rules `position: static; float: none;` won't apply in case of:
- navbars that always collapse `.navbar`
- navbars that never collapse, `.navbar-expand`

This two use cases (navbar that always/never collapse) are already documented [here](https://github.com/twbs/bootstrap/blob/v4-dev/docs/4.0/components/navbar.md#responsive-behaviors).

I mentioned this, first time, [here](https://github.com/twbs/bootstrap/pull/22644#discussion_r117097343). I hope now It's much more clear :)